src/handlers/outgoing-payment.js

```diff
@@ -14,30 +13,21 @@ async function getOutgoingPaymentGrant(deps, session, args) {
     receiveAmount
   );
 
-  const interactRef = await getInteractRef(deps, session);
+  await sleep(2000); // Assume it will take a few seconds to navigate to the grant screen
 
-  if (!interactRef) {
-    return;
-  }
-
-  session.vars.INTERACT_REF = interactRef;
-
-  const grant = await continueOutgoingPaymentGrant(deps, session);
-  return grant;
-}
-
-async function getInteractRef(deps, session) {
-  const urlWithInteractRef = await inquirer.input({
-    message: "Enter URL...",
-    validate: (input) =>
-      !!querystring.parse(input).interact_ref || input === "exit",
-  });
-
-  if (urlWithInteractRef === "exit") {
-    return;
+  try {
+    const approvedPaymentGrant = await poll({
+      request: () => continueOutgoingPaymentGrant(deps, session),
+      timeoutMs: 30000,
+      pollingFrequencyMs: 1000,
```

Comment:
Should we use the `continue.wait` property that we receive from the initial grant request as the polling frequency?

Reply:
Yep, good call, I forgot we get that back in the intial request as well :+1: